test/powershell/Language/Parser/TypeAccelerator.Tests.ps1

```diff
@@ -1,25 +1,437 @@
+                @{
+                    Accelerator = 'powershell'
+                    Type        = [System.Management.Automation.Powershell]
+                }
+                @{
+                    Accelerator = 'runspacefactory'
+                    Type        = [System.Management.Automation.Runspaces.RunspaceFactory]
+                }
+                @{
+                    Accelerator = 'runspace'
+                    Type        = [System.Management.Automation.Runspaces.Runspace]
+                }
+                @{
+                    Accelerator = 'initialsessionstate'
+                    Type        = [System.Management.Automation.Runspaces.InitialSessionState]
+                }
+                @{
+                    Accelerator = 'psscriptmethod'
+                    Type        = [System.Management.Automation.PSScriptMethod]
+                }  
+                @{
+                    Accelerator = 'psscriptproperty'
+                    Type        = [System.Management.Automation.PSScriptProperty]
+                }
+                @{
+                    Accelerator = 'psnoteproperty'
+                    Type        = [System.Management.Automation.PSNoteProperty]
+                }
+                @{
+                    Accelerator = 'psaliasproperty'
+                    Type        = [System.Management.Automation.PSAliasProperty]
+                }
+                @{
+                    Accelerator = 'psvariableproperty'
+                    Type        = [System.Management.Automation.PSVariableProperty]
+                }
+            )
+        
+            if ( $IsCoreCLR )
+            {
+                $totalAccelerators = 89 
+            } else
+            {
+                $totalAccelerators = 94
 
-    It "Basic type accelerator usage" {
-        [msft_2174855] | Should Be ([int])
-    }
+                $nonCoreTypeAcceleratorTestCases = @(
+                    @{
+                        Accelerator = 'adsi'
+                        Type        = [System.DirectoryServices.DirectoryEntry]
+                    }
+                    @{
+                        Accelerator = 'adsisearcher'
+                        Type        = [System.DirectoryServices.DirectorySearcher]
+                    }
+                    @{
+                        Accelerator = 'wmiclass'
+                        Type        = [System.Management.ManagementClass]
+                    }
+                    @{
+                        Accelerator = 'wmi'
+                        Type        = [System.Management.ManagementObject]
+                    }
+                    @{
+                        Accelerator = 'wmisearcher'
+                        Type        = [System.Management.ManagementObjectSearcher]
+                    }
+                )
+            }
+        }
+
+        It 'Should have all the type accelerators' {
+            $TypeAccelerators.Count | Should be $totalAccelerators
+        }

```

Comment:
Yes, I see your point. In `DefaultCommands.Tests.ps1` we guarantee no unexpected `alias` or `cmdlet` by comparing the full list and that's why we don't need the count check in that test.